share/doc/wake/logging.md

```diff
@@ -0,0 +1,71 @@
+# Logging
+
+This article gives a high level overview of how wake logging works.
+Check out [print.wake](https://github.com/sifive/wake/blob/master/share/wake/lib/core/print.wake)
+for commented code.
+
+## How Logging works on Wake Execution commands
+
+Every `Job` run by wake will have its own stdout and stderr,
+and when the underlying process prints to stdout or stderr,
+it will go the `Job`-specific stdout and stderr,
+not the stdout and stderr of the `wake` process itself.
+
+When writing your wake code,
+you can control whether these `Job`-specific stdout and stderr are also printed to `wake`'s own stdout,
+stderr, or other arbitrary file descriptors.
+These can be controlled with `setPlanStderr` and `setPlanStdout` before you `runJob`.
+We call this which _logger_ each goes to (wake code often refers to a _logger_ as a `logLevel`).
+
+Regardless of to which logger you direct the `Job`'s `Stderr` and `Stdout`,
```

Comment:
One more thing which may or may not be worth mentioning here, you can also call `setPlanEcho` with a LogLevel to change where and how noisily the command line/label is printed.  That's generally of much less utility, but it's broadly the same mechanisms involved.

Reply:
> how noisily the command line/label 

Good to know! I will take a look at that too.

Reply:
added this information in parallel with stdout and stderr